static/app/utils/reactQuery.tsx

```diff
@@ -0,0 +1,70 @@
+import * as reactQuery from '@tanstack/react-query';
+import {QueryClientConfig} from '@tanstack/react-query';
+
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+type QueryKeyEndpointOptions = {
+  query?: Record<string, any>;
+};
+
+type QueryKey = readonly [string] | readonly [string, QueryKeyEndpointOptions];
+
+type UseQueryOptions<TQueryFnData, TError, TData, TQueryKey extends QueryKey> = Omit<
```

Comment:
small it, we don't use `T` prefixs very often in our codebase I think since you can just tell based on the hover hints that it's a type

Reply:
Hmm - I quite like the `T` prefix to differentiate generic type arguments from concrete types. Like here I do `TQueryKey extends QueryKey` which I think does a good job of describing which is which (and means I don't have to name `QueryKey` something else just to avoid the name collision.

Reply:
I liked your argument @malwilley :)